src/hotspot/cpu/riscv/macroAssembler_riscv.hpp

```diff
@@ -678,6 +678,12 @@ class MacroAssembler: public Assembler {
   int pop_v(unsigned int bitset, Register stack);
 #endif // COMPILER2
 
+  bool in_range_auipc_s12(int64_t x) {
+    constexpr int64_t twoG = (2 * G);
+    constexpr int64_t twoK = (2 * K);
+    return x <= (twoG - twoK) && x >= (-twoG - twoK);
```

Comment:
Hmm.. I think the range should be `x < (twoG - twoK) && x >= (-twoG - twoK)` which maps to `[-(2G + 2K), 2G - 2K)`.

Reply:
Sorry, yea, the max range is:
2147481599/0x7FFFF7FF and (twoG - twoK) is 2147481600.
-2147485696/0xFFFFFFFF7FFFF800 (0xFFFFFFFF80000000 -  0xFFF) and (-twoG - twoK) is -2147485696.